Move remainder of Angular app under frontend

myabc opened the pull request:
Templates, project infrastructure, configuration.

See 6ec5ca58, #2261.

Also handles:
https://community.openproject.org/work_packages/17538

Continuation of code reorganisation started in #2193. This PR will another step towards allowing us  to move the frontend app into a separate repo.

myabc committed:
Move remainder of Angular app under frontend

Templates, project infrastructure, configuration.

See 6ec5ca58.

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Fix Webpack naming collision running Karma tests

This is the *actual* fix for the issue mentioned in 6ec5ca58.

Signed-off-by: Alex Coles <alex@alexbcoles.com>

TeatroIO commented:
I've prepared a stage. <a href='http://feature-frontend-reorganisation-complete.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Click to open.</a>

myabc committed:
Revert "Invoke bower:install on Travis CI runs"

See 1cd4d68d.

This reverts commit 9bf1265627ef8eb73cd45bdaa0f05f3f6347bbf7.

Signed-off-by: Alex Coles <alex@alexbcoles.com>

Conflicts:
	lib/tasks/ci.rake

myabc committed:
Update paths in Rake task for Webpack compilation

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Fix paths in Rake assets:clobber task

Changed in previous commit 3d34ff2e.

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Update paths for assets process in Procfile

Changed in previous commit 25a6b61f.

myabc committed:
Bundle all (Bower) dependencies with Webpack

Build a separate bundle for "global" dependencies shared between the
classic Rails application and Angular app.

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Move moment-timezone data under frontend/vendor

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Replace jQuery.noop usages w/Lo-Dash equivalent

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Add, configure Webpack loaders for styles, files

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Bundle styles, assets for dependencies w/ Webpack

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Remove a no longer necessary stylesheet file

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Fix stylesheet path used by Express

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Add Function#bind polyfill to placate PhantomJS

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Add Webpack loaders for Angular templates

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Add skeleton README

[ci skip]

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Configure Webpack to bundle Angular HTML templates

User Story # 17538

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Remove karma-ng-html2js-preprocessor

As of 21b28246, Webpack now handles bundling HTML templates.

This reverts commit 73ff6ad52b1b59309bf1c216f5bbe6e37f82334f

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Update templates module in Karma tests

Changed in previous commit 21b28246.

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Merge branch 'dev' into feature/frontend-reorganisation-complete

Signed-off-by: Alex Coles <alex@alexbcoles.com>

Conflicts:
	frontend/tests/integration/index.html

myabc committed:
Bundle angular-busy templates with Webpack

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Tell Teatro to npm (and not bower) install

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Make gulp Express task async

Signed-off-by: Alex Coles <alex@alexbcoles.com>

0xF013 commented:
@myabc I think it happens in me 1.3 branch as well

myabc commented:
@0xF013 I also added template caching in this branch (albeit with Webpack rather than gulp: see commit 6f17d51f). I guess the Protractor failure might be related to this (i.e. many fewer HTTP requests being made).

myabc committed:
Bundle angular-i18n locale data with Webpack

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Tell Teatro to npm install with --unsafe-perm

postinstall `bower install` needs additional permissions.

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Configure Webpack public path

Necessary for assets that Webpack does not inline.

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Remove template paths from Express config

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Attempt fix for query menu items cuke

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Load only angular-i18n locale data for de-DE

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Remove files from non-stupid-digest-assets config

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Tell git to ignore all Webpack-bundled files

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Remove moved files from JSHint ignore config

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc commented:
Also deals with:
https://community.openproject.org/work_packages/17538

myabc committed:
Rename npm package to openproject-frontend

Signed-off-by: Alex Coles <alex@alexbcoles.com>

myabc committed:
Create root package.json aliasing scripts

This should ensure CI and pkgr.io continue working as expected.

Signed-off-by: Alex Coles <alex@alexbcoles.com>

0xF013 commented on app/assets/javascripts/application.js.erb:
```diff
@@ -10,21 +10,10 @@
+
+//= require ./bundles/openproject-global
+
+//# require jquery_noconflict
```
is this commented out?

myabc replied:
@0xF013 it is indeed. In fact, we can probably safely remove `jquery_noconflict`. Webpack only exposes `jQuery` as a global.
